tcc/src/main/java/com/alibaba/fescar/rm/tcc/interceptor/ActionInterceptorHandler.java

```diff
@@ -0,0 +1,203 @@
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.alibaba.fescar.rm.tcc.interceptor;
+
+import com.alibaba.fastjson.JSON;
+import com.alibaba.fescar.common.Constants;
+import com.alibaba.fescar.common.exception.FrameworkException;
+import com.alibaba.fescar.common.executor.Callback;
+import com.alibaba.fescar.common.util.NetUtil;
+import com.alibaba.fescar.core.context.RootContext;
+import com.alibaba.fescar.core.model.BranchType;
+import com.alibaba.fescar.rm.DefaultResourceManager;
+import com.alibaba.fescar.rm.tcc.api.BusinessActionContext;
+import com.alibaba.fescar.rm.tcc.api.BusinessActionContextParameter;
+import com.alibaba.fescar.rm.tcc.api.TwoPhaseBusinessAction;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Handler the TCC Participant Aspect : Setting Context, Creating Branch Record
+ *
+ * @author zhangsen
+ */
+public class ActionInterceptorHandler {
+	
+    private static final Logger LOGGER = LoggerFactory.getLogger(ActionInterceptorHandler.class);
+
+    /**
+     * Handler the TCC Aspect
+     *
+     * @param method         the method
+     * @param arguments      the arguments
+     * @param businessAction the business action
+     * @param targetCallback the target callback
+     * @return map map
+     * @throws Throwable the throwable
+     */
+    public Map<String, Object> proceed(Method method, Object[] arguments, TwoPhaseBusinessAction businessAction, Callback<Object> targetCallback) throws Throwable {
+		Map<String, Object> ret = new HashMap<String, Object>(16);
+		
+		//TCC name
+        String actionName = businessAction.name();
+        String xid = RootContext.getXID();
+        BusinessActionContext actionContext = new BusinessActionContext();
+        actionContext.setXid(xid);
+        //set action anme
+        actionContext.setActionName(actionName);
+        //TODO fescar当前版本暂无主事务记录上下文
+        
+        //Creating Branch Record
+        String branchId = doTccActionLogStore(method, arguments, businessAction, actionContext);
+        actionContext.setBranchId(branchId);
+        
+        //set the parameter whose type is BusinessActionContext
+        Class<?>[] types = method.getParameterTypes();
+        int argIndex = 0;
+        for (Class<?> cls : types) {
```

Comment:
why need BIZ  specifies the parameter location instead of placing it at the end?

Reply:
BusinessActionContext param

Reply:
> why need BIZ specifies the parameter location instead of placing it at the end?

@slievrly We don't specify the location of  BusinessActionContext , user can  put it anywhere.